zend_abstract_interface/config/config_decode.c

```diff
@@ -105,49 +105,100 @@ static bool zai_config_decode_int(zai_str value, zval *decoded_value) {
     return true;
 }
 
-static bool zai_config_decode_map(zai_str value, zval *decoded_value, bool persistent) {
+static bool zai_config_decode_map(zai_str value, zval *decoded_value, bool persistent, bool lowercase, bool map_keyless) {
     zval tmp;
     ZVAL_ARR(&tmp, pemalloc(sizeof(HashTable), persistent));
     zend_hash_init(Z_ARRVAL(tmp), 8, NULL, persistent ? ZVAL_INTERNAL_PTR_DTOR : ZVAL_PTR_DTOR, persistent);
 
     char *data = (char *)value.ptr;
     if (data && *data) {  // non-empty
         const char *key_start, *key_end, *value_start, *value_end;
+
         do {
-            if (*data != ',' && *data != ' ' && *data != '\t' && *data != '\n') {
-                key_start = key_end = data;
-                while (*++data) {
+            while (*data == ',' || *data == ' ' || *data == '\t' || *data == '\n') {
+                data++;
+            }
+
+            if (*data) {
+                key_start = data;
+                key_end = NULL;
+                bool has_colon = false;
+
+                while (*data && *data != ',') {
                     if (*data == ':') {
-                        while (*++data && (*data == ' ' || *data == '\t' || *data == '\n'))
-                            ;
+                        has_colon = true;
+                        data++;
+                        
+                        while (*data == ' ' || *data == '\t' || *data == '\n') data++;
 
-                        value_start = value_end = data;
-                        if (!*data || *data == ',') {
-                            --value_end;  // empty string instead of single char
+                        if (*data == ',' || *data == '\0') {
+                            value_start = value_end = NULL;
                         } else {
-                            while (*++data && *data != ',') {
+                            value_start = data;
+                            value_end = value_start;
```

Comment:
```suggestion
                            value_start = value_end = data;
```
I think this communicates more clearly "we assign both to data", than start to data and end to start.

Reply:
Thank you! Lol learned about this in C. Applied in: https://github.com/DataDog/dd-trace-php/pull/2922/commits/0de558f64ef4a156ba377c3ea35f7a69017bad86